apps/molgenis-components/src/client/client.ts

```diff
@@ -264,13 +264,15 @@ const fetchSchemaMetaData = async (
   axios: Axios,
   schemaName?: string
 ): Promise<ISchemaMetaData> => {
-  if (schemaName && schemaCache.has(schemaName)) {
-    return schemaCache.get(schemaName) as ISchemaMetaData;
+  const currentSchemaName = schemaName ? schemaName : "CACHE_OF_CURRENT_SCHEMA";
+  if (schemaCache.has(currentSchemaName)) {
+    return schemaCache.get(currentSchemaName) as ISchemaMetaData;
   }
   return await axios
     .post(graphqlURL(schemaName), { query: metaDataQuery })
     .then((result: AxiosResponse<{ data: { _schema: ISchemaMetaData } }>) => {
       const schema = result.data.data._schema;
+      schemaCache.set(currentSchemaName, schema);
```

Comment:
should this always be set, or only when no schemaname was provided. I'm a bit worried about tables with cross schema references (reference schema replacing the one you want cached).

Reply:
it doesn't really matter, because in practice its value is either the special 'current_schema' or an actual schema name.
But I will add an 'if' statement to make you feel better :-)